### PR TITLE
Support for handling cookies between requests

### PR DESCRIPTION
**Pull request by danieldotnl**

### CODE REVIEW

File: custom_components/multiscrape/coordinator.py [hunk 1]
```diff
@@ -68,7 +65,7 @@ async def get_content(self) -> str:
 
         if self._form_submitter:
             try:
-                result = await self._form_submitter.async_submit(resource)
+                result, self._cookies = await self._form_submitter.async_submit(resource)
```
Comment (jeremicmilan) [inline custom_components/multiscrape/coordinator.py:68]: Consider defining a class for form_submit result.

**Comment by stormshaker** [comment]
I have a website that sets a session ID in a cookie in the form response header. I could test this branch if you created a Pre-Release of it? Is there anything special needed in a config, or will it submit all cookies back with the page request, like a browser?

**Comment by danieldotnl** [comment]
I will release this soon. No config required, it will indeed send all cookies back.

**Comment by stormshaker** [comment]
I tried testing this PR. The cookies look strange in the log_response files—is that just how the library formats the log, or is it the way the cookie is extracted?

From the form response, I expect a single cookie: __Host-secureSessionID. It's the auth token after a successful login. If I look at the form_submit_response_cookies.txt, it looks like this (token redacted).

```
<Cookies[<Cookie __Host-secureSessionID=ad44[REDACTED]aba for app.boxdivvy.com.au />, <Cookie remember=n for app.boxdivvy.com.au />]>
```

Does that look right? Is it supposed to have the 'for app.boxdivvy.com.au'? The page_request_cookies.txt file is identical. My page_response_body.txt I'm getting back is just the login page again.

**Comment by danieldotnl** [comment]
Thanks for testing this PR! It's not clear to me if everything is working fine and you are just worried about the format of the cookies, or if there is an actual problem?
Not sure why you expect one cookie. If you check the browser, you'll see two cookies, just like logged in the multiscrape logging file:
image

**Comment by stormshaker** [comment]
No, not worried there are two. Just asking if the gurney is as expected, the 'for web.site.name' looked strange, but that could just be how they're displayed. Not working yet, I'll keep trying.

**Pull request merged**